Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/fragments/EmulationFragment.java

```diff
@@ -1,9 +1,11 @@
 package org.dolphinemu.dolphinemu.fragments;
```

Comment:
I'm not seeing seeing the benefit of this commit. It has a ton of complexity, callbacks, etc, that don't seem to add anything.

The current implementation sets up Sys in the emulation thread, right? That's already async. How long does this process take?

A better idea would be to do this work when the Application starts. Then we won't ever have to block in the emulation thread.

Reply:
> The current implementation sets up Sys in the emulation thread, right? That's already async.

No, it's on a completely separate thread. But it is true that the extraction already is async.

> How long does this process take?

I haven't done any proper measurements, but when testing PR #6051, it seemed to take a few seconds.

Reply:
@hackbar current implementation also doesn't really check if the extraction is done or not, if the extraction time was long and the user starts any game we will have a failure. 

However I do agree with you it would be a lot easier if we start the process on app startup and block the user from navigating the app until we finish extraction. It would make the code simpler. The reason for not doing this was mentioned in the original PR [here](https://github.com/dolphin-emu/dolphin/pull/6051#issuecomment-328975765)

Reply:
To clarify, the extraction itself is async, but PR #6051 makes the main thread block on the extraction thread if you try to access the settings or launch emulation before the extraction is done. The intention of the last commit is to avoid that – we still block the user from interacting with the UI, but the main thread itself is no longer blocked, so there's no risk of getting "application not responding". I do agree that it adds complexity – what's the right way to go with this isn't totally clear, as you can see in the comments on PR #6051.